diag_manager/fms_diag_yaml.F90

```diff
@@ -62,9 +65,19 @@ module fms_diag_yaml_mod
 !> @{
 contains
 
+!> @brief gets the diag_yaml module variable
+!! @return a copy of the diag_yaml module variable
+function get_diag_yaml_obj() &
+result(res)
+  type (diagYamlObject_type) :: res
+
+   res = diag_yaml
+end function get_diag_yaml_obj
```

Comment:
This was added for the test. 
The test will get a copy for the **private** `diag_yaml` module variable, and then use the get_* functions to check for the correct values.